### PR TITLE
fix(core/webidl): find interface dfn in section

### LINKED ISSUE #1789: <dfn> for Gamepad is being ignored
## Important info

* URL to affected spec: http://w3c.github.io/gamepad/#gamepad-interface
* ReSpec version: 22.3.3
* [x] I did a "hard refresh", but it's still busted. 

## Description of problem

What happened (e.g., it crashed)?: 

image

But what I was expecting?: `Gamepad` should get a correct link.

This looks similar with #1768 but not sure whether the same bug is causing this.

### PR DESCRIPTION
**Pull request by marcoscaceres**
closes #1789, #1768, #1794

### CODE REVIEW

**Comment by marcoscaceres** [comment]
This now finds the interface DFN in a case sensitive manner, but without needing to change the behavior of `definitionMap`.

File: src/core/webidl.js [hunk 1]
```diff
@@ -664,6 +664,11 @@ function findDfn(parent, name, definitionMap, type, idlElem) {
     // an explicitly empty [for], try <dfn> that inherited a [for].
     if (dfns.length === 0 && parent === "" && dfnForArray.length === 1) {
       dfns = dfnForArray;
+    } else if (type === "interface" && dfnForArray.length) {
```
Comment (saschanaz) [inline src/core/webidl.js]: Why not `interface mixin`, `dictionary`, `namespace`, `typedef`, etc. but only interface?

Comment (marcoscaceres) [reply src/core/webidl.js]: This is why we pay you the big bucks, @saschanaz! :) yes, it should check any top level container type.

File: src/core/webidl.js [hunk 2]
```diff
@@ -33,6 +33,15 @@ const idlTypedefTmpl = tmpls["typedef.html"];
...
 const unlinkable = new Set(["maplike", "setlike", "stringifier"]);
+const topLevelEntities = new Set([
+  "callback interface",
+  "callback",
+  "dictionary",
+  "enum",
+  "interface mixin",
+  "interface",
+  "typedef",
+]);
```
Comment (saschanaz) [inline src/core/webidl.js]: We also have `namespace`s, but it seems we don't support namespaces yet! I didn't know that.

File: js/core/templates/webidl-contiguous/typedef.html [hunk 3]
```diff
@@ -1,2 +1,2 @@
-<span class='idlTypedef' id='{{obj.idlId}}' data-idl data-title='{{obj.name}}'>typedef<span class='idlTypedefType'>{{idlType obj
-}}</span> <span class='idlTypedefID'>{{#tryLink obj}}{{obj.name}}{{/tryLink}}</span>;</span>
\ No newline at end of file
+<span class='idlTypedef' id='{{obj.idlId}}' data-idl data-title='{{obj.name}}'>{{{trivia obj.trivia.base}}}typedef<span class='idlTypedefType'>{{idlType obj
```
Comment (marcoscaceres) [inline js/core/templates/webidl-contiguous/typedef.html]: @saschanaz, is `termination` ok here? Also, I was unsure where to put `trivia.name`? See also https://github.com/w3c/webidl2.js/issues/225

File: src/core/webidl.js (hunk 2 above)
Comment (marcoscaceres) [reply src/core/webidl.js]: Let's do that as a new PR.

**Comment by marcoscaceres** [comment]
also closes #1794

File: src/core/webidl.js (hunk 2 above)
Comment (marcoscaceres) [reply src/core/webidl.js]: Filed https://github.com/w3c/respec/issues/1795

File: js/core/templates/webidl-contiguous/typedef.html (hunk 3 above)
Comment (saschanaz) [reply js/core/templates/webidl-contiguous/typedef.html]: >is termination ok here?

Yep.

> I was unsure where to put trivia.name?

Probably before `<span class='idlTypedefID>`, replacing the existing space character.

File: src/core/webidl.js [hunk 4]
```diff
@@ -457,7 +457,7 @@ function writeIterableLike(iterableLike) {
   const { type, readonly } = iterableLike;
   return idlIterableLikeTmpl({
     obj: iterableLike,
-    qualifiers: readonly ? `${writeTrivia(readonly.trivia)}readonly ` : "",
+    qualifiers: readonly ? `${writeTrivia(readonly.trivia)}readonly` : "",
```
Comment (marcoscaceres) [inline src/core/webidl.js]: Much better!

File: js/core/templates/webidl-contiguous/iterable-like.html [hunk 5]
```diff
@@ -1,3 +1,3 @@
-<span class='{{className}}' id="{{obj.idlId}}" data-idl data-title='iterable'>{{{trivia
-obj.trivia.base}}}{{extAttr obj}}{{qualifiers}}{{#tryLink obj}}{{obj.type}}{{/tryLink
-}}&lt;{{idlType obj}}&gt;;</span>
\ No newline at end of file
+<span class='{{className}}' id="{{obj.idlId}}" data-idl data-title='iterable'
+>{{{qualifiers}}}{{{trivia obj.trivia.base}}}{{#tryLink obj}}{{obj.type}}{{/
```
Comment (marcoscaceres) [inline js/core/templates/webidl-contiguous/iterable-like.html]: 🙇‍♂️

File: js/core/templates/webidl-contiguous/typedef.html (hunk 3 above)
Comment (marcoscaceres) [reply js/core/templates/webidl-contiguous/typedef.html]: Ok, trivia for name is in!

**Comment by marcoscaceres** [comment]
Now uses the separator + trivia in the IDL object structure too - so we have better trivia support throughout.

**Pull request merged**